app/views/directives/create-secret.html

```diff
@@ -292,7 +292,7 @@
   <div class="buttons gutter-top-bottom">
     <button class="btn btn-lg btn-primary"
             type="button"
-            ng-disabled="secretForm.$invalid || secretForm.$pristine"
+            ng-disabled="secretForm.$invalid || secretForm.$pristine || invalidConfigFormat"
```

Comment:
We should be calling `$setValidity` in the controller to mark the field invalid, and then just check `$invalid` here as before. For instance, give the input `name="dockerConfig"`, then in the controller (untested)

```js
        var updateEditorMode = _.debounce(function(){
          try {
            JSON.parse($scope.newSecret.data.dockerConfig);
            $scope.$apply(function() {
              $scope.secretForm.dockerConfig.$setValidity('validJSON', true);
            });
          } catch (e) {
            $scope.$apply(function() {
              $scope.secretForm.dockerConfig.$setValidity('validJSON', false);
            });
          }
        }, 300, {
          'leading': true
        });
```

Use `$scope.$apply` because the function is debounced.

In the view, show the error with

```html
        <div class="has-warning" ng-if="secertForm.dockerConfig.$error.validJSON">
          <span class="help-block">
            Configuration file should be valid JSON.
          </span>
        </div>
```